packages/flutter/lib/src/widgets/focus_manager.dart

```diff
@@ -1873,6 +1893,16 @@ class FocusManager with DiagnosticableTreeMixin, ChangeNotifier {
     assert(_focusDebug(() => 'Notified ${_dirtyNodes.length} dirty nodes:', () => _dirtyNodes));
     _dirtyNodes.clear();
     if (previousFocus != _primaryFocus) {
+      if (_primaryFocus != null && (_primaryFocus!.context?.mounted ?? false)) {
+        final int? viewId = View.maybeOf(_primaryFocus!.context!)?.viewId;
```

Comment:
IMO `_lastFocusedViewId` should be updated here, not by using the engine events. This is because we have two distinct concepts of "focus":

1. Flutter Focus:  Managed by the framework, this represents the focus state within the Flutter widget tree.

1. Platform Focus: Managed by the platform (iOS, Android, Web), this represents the actual focus state of the "native" views.

These two can diverge, but we reconcile them using the view focus methods in the platform dispatcher. Updating `_lastFocusedViewId` here aligns with the idea that we're working with the Flutter focus concept.